src/haxelib/client/Main.hx

```diff
@@ -154,27 +154,27 @@ class Main {
 		args = Sys.args();
 
 		commands = new List();
-		addCommand("install", install, "install a given library, or all libraries from a hxml file", Basic);
-		addCommand("update", update, "update a single library (if given) or all installed libraries", Basic);
+		addCommand("install", checkUpdateThen.bind(install), "install a given library, or all libraries from a hxml file", Basic);
```

Comment:
I think this should be done differently. We have `net` property for commands, so the check should be simply added after `loadProxy` here: https://github.com/HaxeFoundation/haxelib/blob/development/src/haxelib/client/Main.hx#L366